cloudify_nsx/library/nsx_esg_dlr.py

```diff
@@ -167,16 +165,11 @@ def dhcp_server(client_session, esg_id, enabled=None, syslog_enabled=None,
             new_dhcp_config['dhcp']['logging']['logLevel'] = syslog_level
             change_needed = True
 
-    if not change_needed:
-        return True
-    else:
+    if change_needed:
         result = client_session.update('dhcp',
                                        uri_parameters={'edgeId': esg_id},
                                        request_body_dict=new_dhcp_config)
-        if result['status'] == 204:
-            return True
-        else:
-            return False
+        common.check_raw_result(result)
```

Comment:
I have replaced all such "returns" to use check_raw_result - that will raise RecoverableError in case when we can't do something(mainly delete object). Before this change exception will be raised in caller (in case return False).